packages/react-components/react-toast/src/state/useToaster.ts

```diff
@@ -67,13 +119,31 @@ export function useToaster<TElement extends HTMLElement>(options: ToasterProps =
     const cleanupDismissListener = addToastListener(EVENTS.dismiss, dismissToast);
     const cleanupDismissAllListener = addToastListener(EVENTS.dismissAll, dismissAllToasts);
 
+    const focusShortcutListener = (e: KeyboardEvent) => {
+      if (isFocusShortcut(e)) {
+        pauseAllToasts();
+        const mostRecentToast = getMostRecentToast();
+
+        if (mostRecentToast) {
+          lastActiveElementRef.current = isHTMLElement(targetDocument.activeElement)
```

Comment:
Wondering how this will work with Shadow DOM. Also, wondering why `isHTMLElement()` check is needed?

Reply:
activeElement is not typed as a HTMLElement https://www.typescriptlang.org/play?#code/MYewdgzgLgBApgGzgWzmKAuGAJAKgWQBkBRJVdGAHxjAFcEEYBeGAExGFvKgDoBDYFACWANzikUaKAG4AUDBhA

Reply:
> Wondering how this will work with Shadow DOM

Do you mind clarifying what might not work with shadow DOM, I haven't had the change to work with it regularly

Reply:
> https://codesandbox.io/s/vibrant-smoke-7y6cdl?file=/src/App.js

Yeah, that's out of scope here, we will need to have the same kind of solution in tabster for this ☠️☠️